agent/ecscni/plugin_linux.go

```diff
@@ -83,19 +84,8 @@ func (client *cniClient) setupNS(ctx context.Context, cfg *Config) (*current.Res
 		// Not every netns setup involves ECS Bridge Plugin
 		return nil, nil
 	}
-	if _, err := bridgeResult.GetAsVersion(currentCNISpec); err != nil {
-		seelog.Warnf("[ECSCNI] Unable to convert result to spec version %s; error: %v; result is of version: %s",
-			currentCNISpec, err, bridgeResult.Version())
-		return nil, err
-	}
-	var curResult *current.Result
-	curResult, ok := bridgeResult.(*current.Result)
-	if !ok {
-		return nil, errors.Errorf(
-			"cni setup: unable to convert result to expected version '%v'", bridgeResult)
-	}
 
-	return curResult, nil
+	return current.GetResult(bridgeResult)
```

Comment:
is there a potential err with current.GetResult that we should handle/enrich here?

Reply:
I don't think it will be necessary to enrich this error further than what's already given. 

The simplification to using just the [`GetResult` function](https://github.com/containernetworking/cni/blob/3363d143688bb83ca18489ac8b9dc204c1d49c4a/pkg/types/100/types.go#L64) can potentially return the following errors:
(1) Missing converter error: "no converter for CNI result version %s to %s" ([ref](https://github.com/containernetworking/cni/blob/3363d143688bb83ca18489ac8b9dc204c1d49c4a/pkg/types/internal/convert.go#L55))
(2) A type-assertion error: "failed to convert result" ([ref](https://github.com/containernetworking/cni/blob/3363d143688bb83ca18489ac8b9dc204c1d49c4a/pkg/types/100/types.go#L64))

Those errors would trickle back to the DockerTaskEngine [here](https://github.com/aws/amazon-ecs-agent/blob/883211e24d9255023a68f6351001c291af8a699b/agent/engine/docker_task_engine.go#L1765) and [here](https://github.com/aws/amazon-ecs-agent/blob/883211e24d9255023a68f6351001c291af8a699b/agent/engine/docker_task_engine.go#L1844)